### PR TITLE
chore: docker build optimization

### PR DESCRIPTION
**Pull request by dhhyi**
## PR Type

[x] Build-related changes
[x] CI-related changes

## What Is the Current Behavior?

- `npm run synchronize-lazy-components` slow in docker build
- docker build includes build cache

## What Is the New Behavior?

- docker build is run in CI mode (disables angular build cache and schematic lint fixes)
- read CI variable in schematics from env

## Does this PR Introduce a Breaking Change?

[ ] Yes
[x] No

## Other Information

[AB#74538](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/74538)

### CODE REVIEW

**Comment by jometzner** [comment]
In addition to this PR I want to point out what [The Angular - Persistent disk cache page](https://angular.io/cli/cache) states:

> The Angular CLI checks for the presence and value of the CI environment variable to determine in which environment it is running.

Which is the explanation why no further changes are included in this PR to implement the claim "Disables angular build caches"

**Review by jometzner — APPROVED** [review]

File: angular.json [hunk 1]
```diff
@@ -1,6 +1,13 @@
 {
   "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
   "version": 1,
+  "cli": {
+    "analytics": false,
+    "cache": {
+      "enabled": false,
+      "environment": "ci"
+    }
```
Comment (dhhyi) [inline angular.json:9]: If I follow the documentation here https://angular.io/guide/workspace-config#cache-options this deactivates the cache and sets the cache active for the CI environment.

Comment (jometzner) [reply angular.json:9]: Yes. You're right.

Comment (jometzner) [reply angular.json:9]: I fix this.

**Review by jometzner — APPROVED** [review]

**Review by MaxKless — APPROVED** [review]
:rocket:

**Pull request merged**